lib/internal/fs/rimraf.js

```diff
@@ -174,6 +194,24 @@ function rimrafPromises(path, options) {
 function rimrafSync(path, options) {
   let stats;
 
+  try {
+    stats = statSync(path);
+  } catch (err) {
+    if (err.code === 'ENOENT')
+      throw err;
+  }
+
+  if (!stats.isDirectory()) {
+    throw new ERR_INVALID_ARG_VALUE('path', path, 'is not a directory');
```

Comment:
is this the correct error to throw? unsure of the intent of `ERR_INVALID_ARG_VALUE`

Reply:
I'm not sure that it is. I wanted to throw an `ENOTDIR` error but I'm not sure how to do that. I don't see it happening anywhere else in the code. I tried making a regular `Error` and setting the code to `ENOTDIR` but the linter didn't like that.

Reply:
There doesn't seem to be a convenience function for this error; maybe we should create one.

Reply:
@iansu in `internal/errors.js`, I think we want to add:

```
E('ENOENT',
  'no such file or directory',
  SystemError);
```

You'd then instantiate it something like this:

```js
new SystemError('ERR_SOCKET_BUFFER_SIZE', {syscall: 'rimdir', code: 'ENOENT', path: './', message: 'no such file or directory'', errno: -2})
````

Reply:
whoops, sorry I mean `ENOTDIR`, but I think it should be similar logic.